### PR TITLE
OLS - couple of alternate methods

### PR DESCRIPTION
**Pull request by rjenc29**
A couple of alternative OLS solution methods.

### CODE REVIEW

File: fastats/maths/__init__.py [hunk 1]
```diff
@@ -31,4 +32,6 @@
     't_statistic',
     'relu',
     'softplus',
+    'ols_cholesky',
+    'ols_svd'
```
Comment (pawroman) [inline fastats/maths/__init__.py:36]: Have you done any benchmarking on these? With and without JIT? It's super useful information, would be nice to have a notebook with benchmarks.

File: fastats/maths/ols.py [hunk 2]
```diff
@@ -32,6 +33,42 @@ def ols_qr(A, b):
...
+
+    (R.T @ R) @ x = A.T @ b
+    R.T @ w = A.T @ b
+    R @ x = w
+
+    Find R using Cholesky decomposition.
+    """
+    R = cholesky(A.T @ A)
+    w = solve_triangular(R, A.T @ b, trans='T')
+    return solve_triangular(R, w)
```
Comment (pawroman) [inline fastats/maths/ols.py:53]: Brilliant stuff!

Comment (pawroman) [reply fastats/maths/ols.py:53]: But it won't JIT in the current form, right? `solve_triangular` doesn't seem to be supported in numba: http://numba.pydata.org/numba-doc/0.35.0/reference/numpysupported.html

Comment (rjenc29) [reply fastats/maths/ols.py:53]: Indeed, these won't JIT (same for the existing ols_qr).  We'd need to swap out the factorisation methods with our own at some point to get it to play nicely.

File: fastats/maths/__init__.py (hunk 1 above)
Comment (rjenc29) [reply fastats/maths/__init__.py:36]: They don't JIT, but they have quite different performance characteristics - it's a good point

**Comment by dwillmer** [comment]
This looks awesome!

With regards to performance, you'll find that naive ols is fastest for small /single column data sets, the QR approach is generally very fast for all data sets, and cholesky/SVD are only used in special situations.

Super useful to have these so that they can be swapped out as required.

**Pull request merged**